kernel/src/log_segment.rs

```diff
@@ -514,6 +514,51 @@ pub(crate) struct ListedLogFiles {
     pub latest_crc_file: Option<ParsedLogPath>,
 }
 
+impl ListedLogFiles {
+    fn new(
+        ascending_commit_files: Vec<ParsedLogPath>,
+        ascending_compaction_files: Vec<ParsedLogPath>,
+        checkpoint_parts: Vec<ParsedLogPath>,
+        latest_crc_file: Option<ParsedLogPath>,
+    ) -> Self {
+        // We are adding debug_asserts here, so they don't impact the runtime performance of the released binaries
+        debug_assert!(ascending_compaction_files.windows(2).all(|pair| {
+            let [first, second] = pair else {
+                unreachable!()
+            };
```

Comment:
ah was hoping cargo fmt would let it live on one line...

Reply:
I've noticed it in a few places like e.g. `Self { op, expr }` but maybe this was wishful thinking...